blog/2019-10-07-integrate-infinite-loading/index.md

```diff
@@ -0,0 +1,170 @@
+---
+title: How to integrate Infinite Loading with Gridsome
+slug: infinite-loading-with-gridsome
+author: [codybarr]
+date: 2019-10-07
+tags: ['infinite loading', 'infinite scroll']
+excerpt: 'Infinite loading is an alternative strategy for loading additional nodes/content onto a page without the use of traditional pagination. Infinite loading instead loads the next batch of content when a user scrolls to the bottom of a web page.'
+---
+
+## What is infinite loading? Why should I use it?
+
+Paginating data in Gridsome is easy using the built-in [@paginate directive](/docs/pagination/) and even includes a custom [pager component](/docs/pagination/#pager-component).
+
+Infinite loading provides the same performance as traditional paginatation without requiring any extra navigational UI to be included in a page.
+
+You can check out a full working example here: https://gridsome-infinite-loading.netlify.com
+
+## Install the required dependencies
+
+There are many packages that implement the infinite loading technique but one that works particularly well with Gridsome is [vue-infinite-loading](https://github.com//PeachScript/vue-infinite-loading). To add it to your gridsome project, `cd` into your project's root and run the following:
+
+`yarn add vue-infinite-loading`
+
+## Setup
+
+To use `vue-infinite-loading` you'll need to add it to your `main.js` file first:
+
+```javascript
+import InfiniteLoading from 'vue-infinite-loading'
+
+export default function(Vue, { router, head, isClient }) {
+	Vue.use(InfiniteLoading)
+}
+```
+
+## Blog Example
+
+To demonstrate how to use `vue-infinite-loading` let's assume a pretty typical BlogPost type.
+
+### `<page-query>`
+
+Your paginated `<page-query>` would look something like this:
+
+```graphql
+query Blog($page: Int) {
+	posts: allBlogPost(perPage: 10, page: $page) @paginate {
+		pageInfo {
+			totalPages
+			currentPage
+		}
+		edges {
+			node {
+				id
+				title
+				path
+			}
+		}
+	}
+}
+```
+
+This is a pretty typical way to do traditional pagintation, but we're not going to include a pager component in the UI.
+
+### `<script>`
+
+Next, we'll need to add a few things to our `<script>` tag:
+
+1. A couple of variables in our `data()` method to keep track of our `loadedPosts` as well as the `currentPage`.
+2. Some code to the mounted method to add the initial "page" of data to our `loadedPosts` array.
+3. A handler method to pass to our `vue-infinite-loading` component so it knows how to load the next batch of data.
+
+Here's what that would look like:
+
+```javascript
+// component for displaying our individual blog post previews
+import PostCard from '~/components/PostCard.vue'
+
+export default {
+	components: {
+		PostCard
+	},
+	data() {
+		return {
+			loadedPosts: [],
+			currentPage: 1
+		}
+	},
+	mounted() {
```

Comment:
Could you change this to created() ? :)

Reply:
created() is best to use if there is static content, so it will be included in the markup. mounted() is only added on client-side.

Reply:
Got it. Should be all set 👍